### PR TITLE
multi-Cancel revaultd

### PR DESCRIPTION
**Pull request by darosior**
This implements https://github.com/revault/practical-revault/pull/119.

We adapt all components to use multi-Cancel:
- Sharing and fetching the signatures for all the Cancel transactions
- Monitoring the chain for any of the Cancel transaction
- Changing the various interfaces to have all Cancel transactions instead of a single one

Feedback on the new RPC API would be welcome: i preferred a list of Cancel instead of a mapping from feerate to Cancel.
We still don't chose the Cancel to broadcast depending on the fee estimation in `revault`.

### CODE REVIEW

**Comment by darosior** [comment]
Still TODO:
- Remove `db_cancel_transaction()`
- Check for remaining single-cancel helpers in db/interface

**Comment by darosior** [comment]
Note to self: we'll need to update `revaultd`'s master to `net` and `tx` master before merging this.

**Comment by edouardparis** [comment]
Manually tested with https://github.com/revault/revault-gui/pull/350, seems to be fine

**Comment by darosior** [comment]
Rebased on master

**Comment by darosior** [comment]
Yay, this finally passed CI, i'll re-kick it to be sure it's not flaky.

**Comment by darosior** [comment]
Updated the `miradord` submodule to master.

File: src/commands/utils.rs [hunk 1]
```diff
@@ -223,6 +238,29 @@ pub fn finalized_emer_txs(revaultd: &RevaultD) -> Result<Vec<BitcoinTransaction>
         .map_err(|e| e.into())
 }
 
+/// Get the Unvault transaction for a given vault
+pub fn unvault_tx(
```
Comment (edouardparis) [inline src/commands/utils.rs:242]: I think, in the future we should move these helpers in a `scripts` or `consensus` module with a struct handling the descriptors and the secp_ctx, encapsulating the derivation etc. Revaultd would store this struct and we use it like this: 
`revaultd.consensus.unvault_tx(&vault.derivation_index)` or `revaultd.generator.unvault_tx(&derivation_index)`
(I dont know yet which name)

**Review by edouardparis — APPROVED** [review]
ACK ae94b9a5abce856691335d3586c534f2d309965a
Manually tested with the GUI

**Pull request merged**